test/util/regtest.js

```diff
@@ -0,0 +1,171 @@
+'use strict';
+
+const assert = require('./assert');
+const FullNode = require('../../lib/node/fullnode');
+const SPVNode = require('../../lib/node/spvnode');
+
+const {NodeClient, WalletClient} = require('bclient');
+
+async function initFullNode(options) {
+  const node = new FullNode({
+    prefix: options.prefix,
+    network: 'regtest',
+    apiKey: 'foo',
+    walletAuth: true,
+    workers: true,
+    listen: true,
+    bip37: true,
+    port: options.ports.full.p2p,
+    httpPort: options.ports.full.node,
+    maxOutbound: 1,
+    seeds: [],
+    memory: false,
+    plugins: [require('../../lib/wallet/plugin')],
+    env: {
+      'BCOIN_WALLETHTTPPORT': (options.ports.full.wallet).toString()
```

Comment:
I don't know, this really could be improved. After digging through configuration code this was a workaround to the issue that you can't directly pass options to the plugin, as you can the node.

Reply:
PR #562 includes an alternative solution https://github.com/bcoin-org/bcoin/pull/562 however it may be worthwhile to reconsider how "filtering" of options works for plugins, and to have plugins have there own namespace entirely as if it was another process and application. That may be a separate discussion though.

Reply:
Yeah, there's also this PR to bcfg: https://github.com/bcoin-org/bcfg/pull/2